bin/perlcritic

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash -e
+#
+# Wrap perlcritic to provide custom critic rules transparently.
+
+if ! command -v perlcritic > /dev/null 2>&1; then
+    echo "perlcritic found: make sure you've installed all dependencies."
+    exit 1
+fi
+
+# This script will either be called directly from this project or symlinked from
+# downstream projects.
+
+OS_AUTOINST_COMMONS_DIR=$(dirname $(dirname $(readlink -e "$0")))
```

Comment:
why the double dirname?

Reply:
`readlink $0` will resolve the path of the specific file including symlinks.

that's `a/b/c/os-autoinst-common/bin/perlcritic` but we need what's on `a/b/c/os-autoinst-common/lib/perlcritic`.

First `dirname` deletes `perlcritic`, the second deletes `bin` and we end up with the desired path.

See:

```bash
dirname a/b/c/os-autoinst-common/bin/perlcritic
#> a/b/c/os-autoinst-common/bin
dirname a/b/c/os-autoinst-common/bin
#> a/b/c/os-autoinst-common
```